OpenRA.Mods.Common/HitShapes/Capsule.cs

```diff
@@ -85,6 +85,13 @@ public WDist DistanceFromEdge(WVec v)
 		public WDist DistanceFromEdge(WPos pos, Actor actor)
 		{
 			var actorPos = actor.CenterPosition;
+			var attackablePositions = actor.TraitsImplementing<IAttackablePositions>();
+			if (attackablePositions.Any())
+			{
+				var positions = attackablePositions.First().AttackablePositions(actor);
+				if (positions.Any())
+					actorPos = positions.First();
+			}
```

Comment:
Use `SelectMany` here as well, then use `PositionoClosestTo(actorPos)`? Using `First()` - twice! - here feels a bit dodgy.

Same thing in the hunk below.